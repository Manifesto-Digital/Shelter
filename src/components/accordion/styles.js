import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
