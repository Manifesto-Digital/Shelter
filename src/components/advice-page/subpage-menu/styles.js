import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
