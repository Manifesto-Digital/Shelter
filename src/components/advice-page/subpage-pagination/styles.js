import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
