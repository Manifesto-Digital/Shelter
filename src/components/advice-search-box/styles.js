import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
