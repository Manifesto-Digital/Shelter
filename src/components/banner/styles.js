import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
