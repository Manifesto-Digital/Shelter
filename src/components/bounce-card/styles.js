import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
