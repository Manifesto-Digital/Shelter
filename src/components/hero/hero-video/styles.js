import styled from 'styled-components';
import ReactPlayer from 'react-player';
import { breakpoint } from '../../theme/breakpoint';

export const VideoHero = styled.section`
  overflow: hidden;
  padding-top: 60vh;
  position: relative;
`;

export const VideoWrapper = styled.div``;

export const ImageFallback = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background-image: url(${({ bgImage }) => bgImage});
  background-size: cover;
  background-position: center;
`;

export const Video = styled(ReactPlayer)`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
`;

export const TitleWrapper = styled.div`
  position: absolute;
  left: 50%;
  max-width: 520px;
  top: ${props => props.theme.spacing.xl};
  transform: translateX(-50%);
  width: 100%;
  padding: 0 ${props => props.theme.spacing.standard};
  text-align: center;
  color: ${props => props.theme.palette.white};
`;

export const Title = styled.h1`
  margin-bottom: 0;
  font-size: ${props => props.theme.headers.h1};

  ${breakpoint.tablet`
    font-size: ${props => props.theme.fontsize.hero};
  `}
`;

export const Subtitle = styled.p`
  text-align: center;
  font-size: ${props => props.theme.headers.h2};
`;

export const Red = styled.span`
  color: ${props => props.theme.palette.primary};
`;

export const HeroBanner = styled.div`
  background-color: ${props => props.theme.palette.overlay};
  bottom: 0;
  left: 0;
  padding: ${props => props.theme.spacing.standard};
  position: absolute;
  right: 0;
  width: 100%;
`;

export const ContentContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 0 auto;
  width: 100%;
  max-width: 400px;
`;

export const BannerText = styled.p`
  color: ${props => props.theme.palette.white};
  margin-right: ${props => props.theme.spacing.standard};
  margin-bottom: 0;
  padding-left: ${props => props.theme.spacing.standard};
  border-left: 3px solid ${props => props.theme.palette.white};
  white-space: pre-wrap;

  ${breakpoint.tablet`
    margin-right: ${props => props.theme.spacing.large};
  `}
`;

export const StickyBanner = styled.div`
  background-color: ${props => props.theme.palette.primary};
`;
