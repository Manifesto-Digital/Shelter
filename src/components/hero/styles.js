import styled from 'styled-components'
import { breakpoint } from '../theme/breakpoint'
import { absoluteCenter } from '../styled/utils'

export const HeroNoCard = styled.div`
  width: 100vw;
  position: relative;
  overflow: hidden;
  padding-top: 350px;
  margin-bottom: ${props => props.theme.spacing.padding};
  ${breakpoint.tablet`
    padding-top: 60vh;
`};
`

export const MediaWrapper = styled.div`
  ${absoluteCenter};
  width: 100%;
  height: 100%;
  overflow: hidden;

  img {
    ${absoluteCenter};
    min-width: 100%;
    min-height: 100%;
    width: auto;
    height: auto;
    max-width: none;
    height: 350px;

    ${breakpoint.tablet`
        height: auto;
    `};
  }
`

export const Content = styled.div`
  ${absoluteCenter};
  text-align: center;
  width: 100%;
  padding: 0 ${props => props.theme.spacing.padding};
  color: ${props =>
    props.black
      ? `${props.theme.palette.black}`
      : `${props.theme.palette.white}`};
`

export const Title = styled.h1`
  margin-bottom: 0;
`
export const Subtitle = styled.p`
  font-size: ${props => props.theme.headers.h2};
  margin-bottom: 0;
`
