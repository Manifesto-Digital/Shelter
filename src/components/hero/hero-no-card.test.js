import React from 'react';
import { shallow } from 'enzyme';
import { mountWithTheme, snapshotComponent } from 'test-helpers';
import Hero from './hero-no-card';
import { Title, Subtitle } from './styles';
import { createFactory, createImage } from '../../utils/test-factories';

// Default props
export const createHeroNoCard = createFactory({
  title: 'Mock Title',
  subtitle: 'Mock Subtitle',
  blackText: true,
  image: createImage(),
});

it('renders correctly', () => {
  const mockData = createHeroNoCard();

  snapshotComponent(<Hero content={mockData} />);
});

it('displays the correct title', () => {
  const mockData = createHeroNoCard({ title: 'Mock Title' });

  const wrapper = shallow(<Hero content={mockData} />);
  expect(wrapper.find(Title).text()).toBe(mockData.title);
});

it('displays the correct subtitle', () => {
  const mockData = createHeroNoCard({ subtitle: 'Mock Subtitle' });

  const wrapper = shallow(<Hero content={mockData} />);
  expect(wrapper.find(Subtitle).text()).toBe(mockData.subtitle);
});

it('renders an image', () => {
  const mockData = createHeroNoCard({
    image: createImage({
      file: {
        url: 'https://via.placeholder.com/150',
      },
    }),
  });

  const wrapper = mountWithTheme(<Hero content={mockData} />);
  expect(wrapper.find('img')).toHaveLength(1);
});
