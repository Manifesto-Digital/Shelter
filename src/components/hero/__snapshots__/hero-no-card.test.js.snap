// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  width: 100vw;
  position: relative;
  overflow: hidden;
  padding-top: 350px;
  margin-bottom: 40px;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.c1 img {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  min-width: 100%;
  min-height: 100%;
  width: auto;
  height: auto;
  height: 350px;
  max-width: none;
}

.c2 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  text-align: center;
  width: 100%;
  padding: 0 20px;
  color: #212122;
}

.c3 {
  margin-bottom: 0;
}

.c4 {
  font-size: 1.8em;
  margin-bottom: 0;
}

@media (min-width:40.0625em) {
  .c0 {
    padding-top: 60vh;
  }
}

@media (min-width:40.0625em) {
  .c1 img {
    height: auto;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="Collection hero"
      src="//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?w=1600"
      srcSet="//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?w=800 480w,
        //images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?w=1600"
    />
  </div>
  <div
    className="c2"
  >
    <h1
      className="c3"
    >
      Mock Title
    </h1>
    <p
      className="c4"
    >
      Mock Subtitle
    </p>
  </div>
</section>
`;
